src/tools/sssctl/sssctl_config.c

```diff
@@ -34,6 +34,37 @@
 
 
 #ifdef HAVE_LIBINI_CONFIG_V1_3
+
+static char *sssctl_config_snippet_path(TALLOC_CTX *ctx, const char *path)
+{
+    char *tmp;
+    const char *snippet_dir;
+    const char delimiter = '/';
+    char *dpos;
+
+    tmp = talloc_strdup(ctx, path);
+    if (!tmp) {
+        return NULL;
+    }
+
+    snippet_dir = strrchr(CONFDB_DEFAULT_CONFIG_DIR, delimiter);
+    if (snippet_dir != NULL) {
+        ++snippet_dir;
+    } else {
+        snippet_dir = "conf.d";
+    }
+
+    dpos = strrchr(tmp, delimiter);
+    if (dpos != NULL) {
+        ++dpos;
+        *dpos = 0;
+    } else {
+        *tmp = 0;
+    }
+
+    return talloc_strdup_append(tmp, snippet_dir);
```

Comment:
What if talloc_strdup_append() will return an error?

Reply:
In case of error it returns NULL and it is handled in the caller the same way as mentioned above for talloc_strdup & error message.